File: app/src/main/java/ml/dvnlabs/animize/ui/recyclerview/packagelist/packagelist_holder.java
/*
 * Copyright (c) 2020.
 * Animize Devs
 * Copyright 2019 - 2020
 * Davin Alfarizky Putra Basudewa <dbasudewa@gmail.com>
 * This program used for watching anime without ads.
 *
 */

package ml.dvnlabs.animize.ui.recyclerview.packagelist;

import android.content.Context;
import android.content.Intent;
import android.view.View;
import android.widget.ImageView;
import android.widget.TextView;

import androidx.recyclerview.widget.RecyclerView;

import com.bumptech.glide.Glide;
import com.bumptech.glide.load.engine.DiskCacheStrategy;
import com.bumptech.glide.load.resource.drawable.DrawableTransitionOptions;
import com.bumptech.glide.request.RequestOptions;

import ml.dvnlabs.animize.R;
import ml.dvnlabs.animize.model.playlist_model;
import ml.dvnlabs.animize.ui.activity.StreamActivity;

public class packagelist_holder extends RecyclerView.ViewHolder implements View.OnClickListener {
    private TextView episode;
    private TextView id_anim;
    private TextView title;
    private ImageView thumbnail;

    private playlist_model playlist_model;
    private Context context;

    public packagelist_holder(Context context, View view){
        super(view);
        this.context = context;
        this.episode = (TextView)view.findViewById(R.id.playlist_episode);
        this.id_anim = (TextView)view.findViewById(R.id.playlist_id);
        this.title = (TextView)view.findViewById(R.id.playlist_title);
        this.thumbnail = (ImageView)view.findViewById(R.id.playlist_imgthumb);
        itemView.setOnClickListener(this);
    }

    public void bind_playlist(playlist_model plm){
        this.playlist_model = plm;
        this.title.setText(playlist_model.getTitle());
        this.id_anim.setText(playlist_model.getId_anim());
        String ep = context.getString(R.string.episode_text)+": "+playlist_model.getEpisode();
        this.episode.setText(ep);
        Glide.with(itemView).applyDefaultRequestOptions(new RequestOptions().placeholder(R.drawable.ic_picture_light).error(R.drawable.ic_picture_light)).load(playlist_model.getUrl_image()).transition(new DrawableTransitionOptions().crossFade()).apply(new RequestOptions().diskCacheStrategy(DiskCacheStrategy.ALL).override(600,200).fitCenter()).into(thumbnail);

    }
    @Override
    public void onClick(View v){
        if(this.playlist_model!=null){
            Intent intent = new Intent(context.getApplicationContext(), StreamActivity.class);
            intent.putExtra("id_anim",this.playlist_model.getId_anim());
            context.startActivity(intent);
        }

    }


}

File: app/src/main/java/ml/dvnlabs/animize/ui/recyclerview/packagelist/PackageListHolder.kt
/*
 * Copyright (c) 2020.
 * Animize Devs
 * Copyright 2019 - 2020
 * Davin Alfarizky Putra Basudewa <dbasudewa@gmail.com>
 * This program used for watching anime without ads.
 *
 */
package ml.dvnlabs.animize.ui.recyclerview.packagelist

import android.content.Context
import android.content.Intent
import android.view.View
import android.widget.ProgressBar
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView
import kotlinx.coroutines.CoroutineScope
import kotlinx.coroutines.Dispatchers
import kotlinx.coroutines.launch
import kotlinx.coroutines.withContext
import ml.dvnlabs.animize.R
import ml.dvnlabs.animize.database.AnimizeDatabase
import ml.dvnlabs.animize.model.PlaylistModel
import ml.dvnlabs.animize.ui.activity.StreamActivity
import org.koin.core.KoinComponent
import org.koin.core.inject
import org.koin.core.parameter.parametersOf
import kotlin.math.floor

class PackageListHolder(private val context: Context, view: View) : RecyclerView.ViewHolder(view), View.OnClickListener, KoinComponent {
    private val animizeDB: AnimizeDatabase by inject { parametersOf(context) }
    private val episode: TextView = view.findViewById(R.id.rvText)
    private val progress: ProgressBar = view.findViewById(R.id.rvProgress)
    private var playlist_model: PlaylistModel? = null
    fun bindPlaylist(plm: PlaylistModel?) {
        playlist_model = plm
        val ep = context.getString(R.string.episode_text) + ": " + playlist_model!!.episode
        episode.text = ep
        setupImage(playlist_model!!.id_anim)
    }

    private fun setupImage(anmID: String) = CoroutineScope(Dispatchers.Main).launch {
        withContext(Dispatchers.IO) {
            val recentPlay = animizeDB.recentPlayedDAO().getRecentByAnimeID(anmID)
            var max = 0L
            var current = 0L

            if (recentPlay != null) {
                max = recentPlay.duration
                current = recentPlay.timestamp
                //println("ANMID : ${recentPlay.anmid} TIME : ${recentPlay.timestamp} MAX: ${recentPlay.maxTime}")
            }

            val percent: Int = when (max) {
                0L -> {
                    0
                }

                else -> {
                    floor((current.toDouble() / max.toDouble()) * 100).toInt()
                }
            }
            progress.progress = percent
        }
    }

    override fun onClick(v: View) {
        if (playlist_model != null) {
            val intent = Intent(context.applicationContext, StreamActivity::class.java)
            intent.putExtra("id_anim", playlist_model!!.id_anim)
            context.startActivity(intent)
        }
    }

    init {
        itemView.setOnClickListener(this)
    }
}